src/block_cache/block_mgr.c

```diff
@@ -49,28 +28,16 @@ __bm_close_block(WT_SESSION_IMPL *session, WT_BLOCK *block)
         __wt_spin_unlock(session, &conn->block_lock);
         return (0);
     }
+    __wt_spin_unlock(session, &conn->block_lock);
 
     /* You can't close files during a checkpoint. */
     WT_ASSERT(
       session, block->ckpt_state == WT_CKPT_NONE || block->ckpt_state == WT_CKPT_PANIC_ON_FAILURE);
 
-    /*
-     * Every time we remove a block, we may have sufficiently decremented other references to allow
-     * other blocks to be removed. It's unlikely for blocks to reference each other but it's not out
-     * of the question, either. Loop until we don't find anything to close.
-     */
-    do {
-        found = false;
-        TAILQ_FOREACH (block, &conn->blockqh, q)
-            if (block->ref == 0) {
-                found = true;
-                WT_TRET(__bm_close_block_remove(session, block));
-                break;
-            }
-    } while (found);
-    __wt_spin_unlock(session, &conn->block_lock);
+    if (block->sync_on_checkpoint)
+        WT_RET(__wt_fsync(session, block->fh, true));
```

Comment:
My intuition is that if we are closing a block handle here -- currently the only path is when close the btree -- there can't be unsynced writes here. But we sometimes wind up here with the `sync_on_checkpoint` flag set. I believe this happens because we *always* set that flag which switching files, but if the tree is clean at that point and never gets dirtied, we will never need to checkpoint it, and therefor won't invoke the normal mechanism for clearing this flag in `__bm_checkpoint()`. 

But I can't prove it's safe to ignore it. Nor can I devise and ASSERT that would detect a problem scenario.